torch/_functorch/aot_autograd.py

```diff
@@ -544,10 +544,7 @@ def convert(idx, x):
 
         fake_flat_args = process_inputs(flat_args)
 
-        needs_autograd = (
-            any(x.requires_grad for x in fake_flat_args if isinstance(x, Tensor))
-            and torch.is_grad_enabled()
-        )
+        needs_autograd = any(x.requires_grad for x in fake_flat_args if isinstance(x, Tensor))
```

Comment:
This looks right! Tests please :) you can add them in `test_aotdispatch.py:TestAOTAutograd

Reply:
also cc @jamesjwu - Ivan is working on another silent correctness AOTAutograd fix that boils down to "we traced an inference graph when we should have traced a training graph": https://github.com/pytorch/pytorch/issues/114338

Reply:
Do we need needs_autograd to be defined up here, instead of on line 573? i.e. why don't we refactor both of these if statements into a single check (perhaps in its own function)? 

Right now it's confusing that needs_autograd can be True on 547 and then set to false later, making it temporarily true